### ISSUE TITLE
Folder Configuration Problem 

### ISSUE THREAD

**Issue by Taurean75** [eric-bixby/auto-sort-bookmarks-webext#75]
Hey...

Executing ""Sort by Name" on an excluded sub-folder with the recursive option on resets the configuration for its parent folder.

Thanks!

**Comment by jjaone** [comment]
In Firefoc 103, Folder Configuration does not anything like that (see the screenshot), it opens in a new tab/window where the config can not be saved, so it does not work.
![Screenshot from 2022-08-09 19-46-56](https://user-images.githubusercontent.com/5411149/183711049-dd56f799-bb9b-4243-99e0-4e247d0679d7.png)

**Comment by eric-bixby** [comment]
The "configure folder" preference is based on the UUID of the folder.  When the UUID of the folder changes, then the preference reverts to the default (of sorting rather than being excluded).  Prior to Firefox switching to web-extension API, we were able to store this preference in the "tag" attribute.  I need Firefox to bring back "tags", until then we are stuck with this hack.  I'm thinking about adding the exclude flag to the folder title; this would work but people might be annoyed by the extra characters in the title.

**Comment by eric-bixby** [comment]
This may be related to issue #83

**Comment by Gitoffthelawn** [comment]
Oh, how I would love Firefox to bring back the tags API too! 👍🏾 

Changing the folder title could work... you could even append a narrow non-visible character.  If you go that route, you'll probably want to add clear info about this hack in a prominent place so people don't mistakenly think something is amiss! :)

**Comment by eric-bixby** [comment]
@Gitoffthelawn you are right, adding special characters might work, but would most likely confuse and/or annoy users.  This would be replacing a hack with another hack.

For everyone else not familiar with the history of the current hack.  This issue is related to #54.

Check out this link to a request, by many users, over many years, asking Mozilla to restore/add the "tags" feature to web extensions:
https://bugzilla.mozilla.org/show_bug.cgi?id=1225916

The answer is always "no".  I think this is a political/economic reason why this will never happen (Google is paying for things and they probably don't care about a Firefox exclusive feature):
https://en.wikipedia.org/wiki/Mozilla_Foundation#Financing

Anyway, I might try the title approach as an experiment, but a potentially more elaborate approach would be to export the Firefox bookmarks, then import that into the add-on.  The add-on would then have access to the "tags" attribute.

Of course, the problem with this is that the user would be potentially exporting/importing back and forth.  This might be cumbersome.  I need to try prototyping this approach also.

I'll post betas for both, if anyone wants to try them and give me feedback.

**Comment by eric-bixby** [comment]
I've decided to duplicate/steel how ChromeTag does tagging: https://github.com/rotten77/chromtag
The user will add "#hashtag" to the folder name to indicate exclude-from-sorting or recursive-exclude-from-sort.
The actual tag names still needs to be worked out.  So the web-ui would add/remove the tag.  However, when looking at the bookmark using the built-in bookmark-manager, the user will see the tags.

**Comment by eric-bixby** [comment]
**Triage analysis (2026-09-21)** — Reproduced in Firefox 156, and the cause is in the add-on, not a Firefox UUID change. No tags API or hashtags-in-titles needed for this one.

**Root cause:** `Sorter.sortAllBookmarks()` builds the list of folders to sort by walking the tree and **skipping** any recursively-excluded folder (and its subtree). It then calls `Annotations.removeMissingFolders(allFolders)` to clean up annotations for deleted folders — but that removes the `donotsort`/`recursive` flags for *every* folder ID not in the list, which includes every recursively-excluded folder, because those were deliberately left out. So the first sort after you enable "recursive" on a folder deletes both of its flags, and the next sort sorts its contents. A plain "do not sort" without "recursive" survives because that folder is still collected (it's just skipped by `canBeSorted`).

Trace from the harness (folder S has both flags set and contains `Z Y X`):

```
annotations BEFORE sort #1:  {"donotsort":{"gLKgv…":true},"recursive":{"gLKgv…":true}}
S children after sort #1:    Z Y X                                   (still excluded during this sort)
annotations AFTER  sort #1:  {"donotsort":{},"recursive":{}}         <- wiped
S children after sort #2:    X Y Z                                   <- now sorted
```

Firefox's "Sort by Name" in the original report is just one way to trigger an auto-sort (it fires `onMoved` events); any sort does this. The same logic is in the v3.4.5 release (`getSubFolders` skips recursively-excluded folders, then `tags.removeMissingFolders(mergedFolders)`), so this has been the behavior since the exclude-folders feature shipped. #83 (recursively excluding the toolbar "isn't saved at all") is the same bug.

**Fix direction (not applied yet):** `removeMissingFolders` needs the set of *all existing* folder IDs (e.g. walk `bookmarks.getTree()` without the exclusion filter), not the set of *sortable* ones.
